x/bank/keeper.go

```diff
@@ -8,63 +8,59 @@ import (
 
 const moduleName = "bank"
 
-// CoinKeeper manages transfers between accounts
-type CoinKeeper struct {
-	am sdk.AccountMapper
+func getCoins(ctx sdk.Context, am sdk.AccountMapper, addr sdk.Address) sdk.Coins {
```

Comment:
This extra method indirection is not useful, I think we should just define these as CoinKeeper methods.

Reply:
But then how would I call it from the ViewKeeper for example?

Reply:
I could embed a coin Keeper, but I would prefer to not have to create a new CoinKeeper just to create a ViewKeeper.

Reply:
Can you embed the `ViewKeeper` in the `CoinKeeper` though?

Reply:
But then I have to generate all the keepers when I only want one?

Reply:
I don't like embedding structs because if you have "diamond inheritance" it cries unambiguous selector.  https://play.golang.org/p/-t7jTejvaTg

I think the unexported functions in the module which each keeper can call is cleanest and most versatile without ever having to think about inheritance issues.